rpc/wrtc_signaling_answerer.go

```diff
@@ -303,37 +306,50 @@ func (ans *webrtcSignalingAnswerer) answer(client webrtcpb.SignalingService_Answ
 			return err
 		}
 
-		var callFlowWG sync.WaitGroup
+		var pendingCandidates sync.WaitGroup
 		waitOneHost := make(chan struct{})
 		var waitOneHostOnce sync.Once
-		pc.OnICECandidate(func(i *webrtc.ICECandidate) {
+		pc.OnICECandidate(func(icecandidate *webrtc.ICECandidate) {
 			if exchangeCtx.Err() != nil {
 				return
 			}
-			if i != nil {
-				callFlowWG.Add(1)
-				if i.Typ == webrtc.ICECandidateTypeHost {
+			if icecandidate != nil {
+				pendingCandidates.Add(1)
+				if icecandidate.Typ == webrtc.ICECandidateTypeHost {
 					waitOneHostOnce.Do(func() {
 						close(waitOneHost)
 					})
 				}
 			}
 			// must spin off to unblock the ICE gatherer
 			utils.PanicCapturingGo(func() {
+				ans.activeBackgroundWorkers.Add(1)
```

Comment:
Sorry I missed this before, this `Add` probably needs to be outside the goroutine. I assume that `Wait`ing on `activeBackgroundWorkers` won't happen while inside an `OnICECandidate` callback, but it can happen after `OnICECandidate` returns, but before the goroutine it spawns completes. A small race certainly but this allows for the following interleaving:
```
| OnIceCandidate(Foo) | GoRoutine(Foo)                 | Answerer.activeBackgroundWorkers.Wait() |
|---------------------+--------------------------------+-----------------------------------------|
| PanicCapturingGo    |                                |                                         |
| Return              |                                |                                         |
|                     |                                | Call Wait() // Counter is 0             |
|                     |                                | Return from Wait()                      |
|                     | activeBackgroundWorkers.Add(1) |                                         |
```